axum/src/json.rs

```diff
@@ -112,6 +113,27 @@ where
     }
 }
 
+impl<T, S> OptionalFromRequest<S> for Json<T>
+where
+    T: DeserializeOwned,
+    S: Send + Sync,
+{
+    type Rejection = JsonRejection;
+
+    async fn from_request(req: Request, state: &S) -> Result<Option<Self>, Self::Rejection> {
+        if json_content_type(req.headers()) {
+            let bytes = Bytes::from_request(req, state).await?;
+            if !bytes.is_empty() {
+                Ok(Some(Self::from_bytes(&bytes)?))
```

Comment:
this causes a mismatching JSON body to return an error instead of `None`, is that intentional?

Reply:
is this in reference to the previous CR comment? by mismatch do you mean an empty body here?